### PR TITLE
feat: base structure of theme creator

### PR DESCRIPTION
**Pull request by dgaponov**

### CODE REVIEW

File: src/components/Theme/constants.ts [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    },
+};
+
+export const DEFAULT_PALETTE_TOKENS = new Set(Object.keys(DEFAULT_PALETTE.light));
+
+export const DEFAULT_THEME: ThemeOptions = {
+    palette: DEFAULT_PALETTE,
+    colors: {
+        light: {
+            background: 'private.blue.550',
```
Comment (dgaponov) [inline src/components/Theme/constants.ts:40]: The values may refer to a private color or they can have a specific color value.
The reference processing logic will be implemented in the next PR.

File: src/components/ColorPickerInput/ColorPickerInput.scss [hunk 2]
```diff
@@ -15,6 +15,8 @@ $block: '.#{variables.$ns}color-picker';
         width: 100%;
         height: 0;
         opacity: 0;
-        border: 1px solid transparent;
+        padding: 0;
+        margin: 0;
+        border: 0;
```
Comment (bityutskiyAO) [inline src/components/ColorPickerInput/ColorPickerInput.scss:20]: are u check result in safari and firefox? 
that was a really hard to fix safari problems (if whole height of block become less then 5px - color modal go to different place)

File: src/components/Theme/ColorsTab/BasicPalette/PaletteColors.tsx [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+    onAddColorClick,
+    onDeleteColor,
+    onUpdateColorTitle,
+}) => {
+    return (
+        <Flex direction="column" className={b()} grow={true}>
+            <Text variant="subheader-3" className={b('title')}>
+                Support Colors for various cases and
+                <br /> states
+            </Text>
```
Comment (bityutskiyAO) [inline src/components/Theme/ColorsTab/BasicPalette/PaletteColors.tsx:85]: will u add localization?

File: src/components/Theme/hooks/useThemeCreator.ts [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+    const palette = React.useMemo(() => getThemePalette(state), [state]);
+
+    return {
+        addColor,
+        updateColor,
+        removeColor,
+        renameColor,
+        palette,
+    };
+};
```
Comment (bityutskiyAO) [inline src/components/Theme/hooks/useThemeCreator.ts:57]: u can use useReducer

**Review by bityutskiyAO — CHANGES_REQUESTED** [review]

File: src/components/Theme/hooks/useThemeCreator.ts (hunk 4 above)
Comment (dgaponov) [reply src/components/Theme/hooks/useThemeCreator.ts]: Done

File: src/components/ColorPickerInput/ColorPickerInput.scss (hunk 2 above)
Comment (dgaponov) [reply src/components/ColorPickerInput/ColorPickerInput.scss:24]: I checked it in safari and firefox. Everything is OK.

File: src/components/Theme/constants.ts [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+            hoveredBrand: 'private.yellow.200',
+            brandText: '',
+            hcBrandText: '',
+            brandLine: '',
+            selectionBackground: '',
+            hoveredSelectionBackground: '',
+            link: '',
+            hoveredLink: '',
+            visitedLink: '',
```
[the remark this answers is not in the document]
Comment (dgaponov) [reply src/components/Theme/constants.ts]: Done

File: src/components/Theme/ColorsTab/BasicPalette/ThemePaletteCard.tsx [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+export const ThemePaletteCard: React.FC<ThemePaletteCardProps> = ({theme, palette, onUpdate}) => {
+    const createChangeHandler = React.useCallback(
+        (title: string) => (value: string) => {
+            onUpdate({title, theme, value});
+        },
+        [onUpdate, theme],
+    );
+
+    return (
+        <ThemeProvider theme={theme} rootClassName={b('theme-root', 'sandbox')}>
```
[the remark this answers is not in the document]
Comment (dgaponov) [reply src/components/Theme/ColorsTab/BasicPalette/ThemePaletteCard.tsx]: Done

File: src/components/Theme/constants.ts (hunk 1 above)
Comment (dgaponov) [reply src/components/Theme/constants.ts]: Done in current PR

File: src/components/Theme/ColorsTab/BasicPalette/PaletteColors.scss [hunk 7]
```diff
@@ -0,0 +1,19 @@
+@use '../../../../variables.scss';
+
+$block: '.#{variables.$ns}palette-colors';
+
+#{$block} {
+    --g-button-border-radius: var(--g-spacing-2);
+    --g-text-input-border-radius: var(--g-spacing-2);
```
[the remark this answers is not in the document]
Comment (dgaponov) [reply src/components/Theme/ColorsTab/BasicPalette/PaletteColors.scss]: Fixed

File: src/components/Theme/ColorsTab/BasicPalette/ThemePaletteCard.scss [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+        background: #2c2c2c;
+    }
+
+    &__title {
+        color: var(--g-color-base-background);
+    }
+
+    &__theme-root {
+        --g-color-base-background: transparent;
+        --g-button-border-radius: var(--g-spacing-2);
```
[the remark this answers is not in the document]
Comment (dgaponov) [reply src/components/Theme/ColorsTab/BasicPalette/ThemePaletteCard.scss]: Fixed

**Comment by dgaponov** [comment]
I'm closing this PR. All changes were merged into the PR #216.

**Pull request closed without merging**